src/dao/RestDAO.js

```diff
@@ -0,0 +1,249 @@
+
+  __delete<TQueryParams: Object>(
+    path: string,
+    id: EntityID,
+    queryParams?: TQueryParams,
+  ) {
+    const clientId = ClientID.getClientId();
+    const stringifiedId = id.toString();
+
+    const entity =
+      this._entityLoaderById.get(stringifiedId) || LoadObject.empty();
+    this._entityLoaderById.set(stringifiedId, entity.deleting());
+
+    this._entityLoaderById.set(clientId, LoadObject.empty().deleting());
+    this.__emitChanges();
+
+    fetch(path, { method: 'DELETE', ...queryParams })
+      .then(() => {
+        this._entityLoaderById.delete(id);
+        this._entityLoaderById.delete(clientId);
+        this._flushQueryCaches();
+        this.__emitChanges();
+      })
+      .catch(error => {
+        this._updateCacheForError(clientId, error);
+      });
+
+    return clientId;
+  }
+
+  __fetchMany<TQueryParams: Object>(
+    path: string,
+    queryParams?: TQueryParams,
+  ): LoadObject<Array<LoadObject<TEntity>>> {
+    const cacheKey = this.__getCacheKey(path, queryParams);
+
+    if (!this._entityIdsLoaderByQuery.has(cacheKey)) {
+      this._entityIdsLoaderByQuery.set(cacheKey, LoadObject.loading());
+      this.__emitChanges();
+
+      fetch(path, { method: 'GET', ...queryParams })
+        .then((items: ?Array<TEntity>) => {
+          // todo items should be an array from api but now its null
+          if (!items) {
+            items = []; // eslint-disable-line
+          }
+          const ids = items.map(({ id }) => id);
+
+          items.forEach(item =>
+            this._entityLoaderById.set(item.id, LoadObject.withValue(item)),
+          );
+
+          this._entityIdsLoaderByQuery.set(cacheKey, LoadObject.withValue(ids));
+          this.__emitChanges();
+        })
+        .catch(error => {
+          this._entityIdsLoaderByQuery.set(
+            cacheKey,
+            LoadObject.withError(error),
+          );
+          this.__emitChanges();
+        });
+    }
+
+    return nullthrows(this._entityIdsLoaderByQuery.get(cacheKey)).map(ids =>
+      ids.map(id => nullthrows(this._entityLoaderById.get(id.toString()))),
+    );
+  }
+
+  __fetchOne<TQueryParams: Object>(
+    path: string,
+    id: EntityID,
+    queryParams?: TQueryParams,
+  ): LoadObject<?TEntity> {
+    const stringifiedId = id.toString();
+
+    if (!this._entityLoaderById.has(stringifiedId)) {
+      this._entityLoaderById.set(stringifiedId, LoadObject.loading());
+      this.__emitChanges();
+
+      fetch(path, {
+        headers: [
+          { name: 'Accept', value: 'application/json' },
+          { name: 'Content-Type', value: 'application/json' },
+        ],
+        method: 'GET',
+        ...queryParams,
+      })
+        .then(this._updateCacheForEntity)
+        .catch(error => {
+          this._updateCacheForError(stringifiedId, error);
+        });
+    }
+
+    return nullthrows(this._entityLoaderById.get(stringifiedId));
+  }
+
+  __fetchOnce<TQueryParams: Object>(
```

Comment:
so, i made this for things like ping() , where we need to return new result every time. does it makes sense? Probably not the best method name though.

Reply:
@jlkalberer but is the method logic right? the only way we can refetch every time is like we do for udpate request -using and returning clientID, right?

Reply:
since  if we do it without clientID it will override previous request, which result can be still used somwhere in the UI